subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/ArtifactsTestListener.kt

```diff
@@ -48,7 +50,16 @@ internal class ArtifactsTestListener(
         executionNumber: Int,
         testArtifactsDir: Result<File>
     ) {
-        val tempDirectory = createTempDirectory()
+        val tempDirectory = if (saveTestArtifactsInOutputs) {
+            File(outputDirectory, "test-artifacts").apply {
+                parentFile.mkdirs()
+                require(mkdir()) { "can't mkdir $this" }
+            }.toPath()
+        } else {
+            createTempDirectory()
+        }
```

Comment:
I'd like to avoid "if" on this level.
Can we hide this side effect in a custom file storage implementation?

Reply:
FileStorage is on other level of abstraction, which will requires a lot of work to hide it.